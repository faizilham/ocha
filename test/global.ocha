let count = 0;

fn counter1() {
    count = count + 1;
    print(count);
}

fn makeCounter() {
    fn counter1() {
        let a = 1;
        count = count + a;
        print("inner", count);
    }

    return counter1;
}

let counter2 = makeCounter();

counter1();                         // 1
counter1();                         // 2
counter2();                         // inner 3
counter2();                         // inner 4
counter1();                         // 5
counter2();                         // inner 6
counter1();                         // 7
counter2();                         // inner 8
