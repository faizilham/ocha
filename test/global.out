1
2
inner 3
inner 4
5
inner 6
7
inner 8
